Note different .pubxml handling by VS and CLI (#34269)
dotnet/docs: docs/core/tools/dotnet-publish.md

If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.

> Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties, such as `LastUsedBuildConfiguration`, may never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported.

Reviewer: For docs purposes, I'd want to reach out to @vijayrkn and team and see if there were examples of properties that are 100% known to be irrelevant to the CLI.

Reviewer 2: The most notable ones are properties that represent configuration/platform/Runtime/Target identifiers that impact the build.

`LastUsedBuildConfiguration`
`Configuration`
`Platform`
`TargetFramework`
`TargetFrameworks`
`RuntimeIdentifier`

Author: ```suggestion
Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the ones that impact the build, such as the following properties:

* `LastUsedBuildConfiguration`
* `Configuration`
* `Platform`
* `TargetFramework`
* `TargetFrameworks`
* `RuntimeIdentifier`
```

If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.

> Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the ones that impact the build, such as the following properties:

Author: ```suggestion
Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the following ones that impact the build:
```

* `LastUsedBuildConfiguration`
* `Configuration`

> * `Platform`

Author: ```suggestion
* `Platform`
* `LastUsedPlatform`
```

* `TargetFramework`
* `TargetFrameworks`

> * `RuntimeIdentifier`

Reviewer 2: For completeness, we should also add LastUsedPlatform& RuntimeIdentifiers (https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#runtimeidentifiers) along with RuntimeIdentifier(similar to TargetFrameworks & TargetFramework). Sorry, missing it in my initial email.

> * `RuntimeIdentifier`

Author: ```suggestion
* `RuntimeIdentifier`
*  `RuntimeIdentifiers` 
```

If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.

Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the following ones that impact the build:

* `LastUsedBuildConfiguration`
* `Configuration`
* `Platform`
* `LastUsedPlatform`
* `TargetFramework`
* `TargetFrameworks`
* `RuntimeIdentifier`

> *  `RuntimeIdentifiers` 

Author: ```suggestion
* `RuntimeIdentifiers`
```
